diff --git a/src/org/bestever/bebot/MySQL.java b/src/org/bestever/bebot/MySQL.java
index 1d9167f..e1ab662 100755
--- a/src/org/bestever/bebot/MySQL.java
+++ b/src/org/bestever/bebot/MySQL.java
@@ -316,6 +316,29 @@ public class MySQL {
 			return null;
 		}
 	}
+	
+	/**
+	 * Adds a host to the whitelisted hostmasks list
+	 * @param host String - the hostmask
+	 * @param user String - the owner of the hostmask
+	 */
+	public static void addHost(String host, String user, String sender) {
+		String query = "INSERT INTO `" + mysql_db + "`.`hostmasks` VALUES (NULL, ?, ?) ON DUPLICATE KEY UPDATE `username` = ?";
+		try (Connection con = getConnection(); PreparedStatement pst = con.prepareStatement(query)) {
+			pst.setString(1, host);
+			pst.setString(2, user);
+			pst.setString(3, user);
+			if (pst.executeUpdate() == 1) {
+				bot.sendMessage(sender, "Added " + host + " to whitelisted hostmasks.");
+				bot.sendLogAdminMessage(Colors.BOLD+sender+Colors.BOLD + " whitelists hostmasks " + Colors.BOLD+host+Colors.BOLD + " with owner " + Colors.BOLD+user);
+			}
+			else
+				bot.sendMessage(sender, "That Hostmask address is already whitelisted!");
+		} catch (SQLException e) {
+			e.printStackTrace();
+			logMessage(LOGLEVEL_IMPORTANT, "Could not whitelist Hostmask");
+		}
+	}
 
 	/**
 	 * Adds a ban to the banlist
@@ -323,11 +346,16 @@ public class MySQL {
 	 * @param reason String - the reason to show they are banned for
 	 */
 	public static void addBan(String ip, String reason, String sender) {
-		String query = "INSERT INTO `" + mysql_db + "`.`banlist` VALUES (?, ?) ON DUPLICATE KEY UPDATE `reason` = ?";
+		String date = String.valueOf(System.currentTimeMillis() / 1000L);
+		String query = "INSERT INTO `" + mysql_db + "`.`banlist` VALUES (?, ?, ?, ?) ON DUPLICATE KEY UPDATE `reason` = ?, `banner` = ?, `date` = ?";
 		try (Connection con = getConnection(); PreparedStatement pst = con.prepareStatement(query)) {
 			pst.setString(1, ip);
 			pst.setString(2, reason);
-			pst.setString(3, reason);
+			pst.setString(3, sender);
+			pst.setString(4, date);
+			pst.setString(5, reason);
+			pst.setString(6, sender);
+			pst.setString(7, date);
 			if (pst.executeUpdate() == 1) {
 				bot.sendMessage(sender, "Added " + ip + " to banlist.");
 				bot.sendLogAdminMessage(Colors.BOLD+sender+Colors.BOLD + " adds IP " + Colors.BOLD+ip+Colors.BOLD + " to banlist with reason " + Colors.BOLD+reason);
